components/at_mention/at_mention.tsx

```diff
@@ -1,41 +1,52 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Overlay} from 'react-bootstrap';
 
 import {Client4} from 'mattermost-redux/client';
 import {displayUsername} from 'mattermost-redux/utils/user_utils';
+import {NameMappedObjects, UsernameMappedObjects} from 'mattermost-redux/types/utilities';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Group} from 'mattermost-redux/types/groups';
 
 import ProfilePopover from 'components/profile_popover';
 
-import {popOverOverlayPosition} from 'utils/position_utils.tsx';
+import {popOverOverlayPosition} from 'utils/position_utils';
+
 const spaceRequiredForPopOver = 300;
 
-export default class AtMention extends React.PureComponent {
-    static propTypes = {
-        children: PropTypes.node,
-        currentUserId: PropTypes.string.isRequired,
-        channelId: PropTypes.string,
-        hasMention: PropTypes.bool,
-        disableHighlight: PropTypes.bool,
-        disableGroupHighlight: PropTypes.bool,
-        isRHS: PropTypes.bool,
-        mentionName: PropTypes.string.isRequired,
-        teammateNameDisplay: PropTypes.string.isRequired,
-        usersByUsername: PropTypes.object.isRequired,
-        groupsByName: PropTypes.object.isRequired,
-    };
-
-    static defaultProps = {
+type Props = {
+    children?: React.ReactNode;
+    currentUserId: string;
+    channelId?: string;
+    hasMention?: boolean;
+    disableHighlight?: boolean;
+    disableGroupHighlight?: boolean;
+    isRHS?: boolean;
+    mentionName: string;
+    teammateNameDisplay: string;
+    usersByUsername: UsernameMappedObjects<UserProfile>;
+    groupsByName: NameMappedObjects<Group>;
+}
+
+type State = {
+    show: boolean;
+    target?: EventTarget;
+    placement?: string;
+}
```

Comment:
could you please sort the properties by mandatory and optional? Thanks!

The way it is done in the `State` type is fine